sequencing-server/src/backgroundTranspiler.ts

```diff
@@ -0,0 +1,144 @@
+import { piscina, graphqlClient, typeCheckingCache, promiseThrottler } from './app.js';
```

Comment:
General warning that singleton exports like this make testing hard (ie have to mock inject into the runtime etc). It's probably fine here, but an alternative would be to do dependency injection for these elements.

ie
```
function createBackgroundTranspiler(graphqlClient: <whatever>, promiseThrottler: PromiseThrottler<whatever>, ...etc) {
  return function backgroundTranspiler(numberOfThreads: number = 2) {
    // Body of function
  }
}
```

and then initialize it before using it in the context where you have all the dependencies for injection.

Reply:
Thank you for the suggestion, I noticed are doing this singleton export throughout the server which we probably need to refactor. Let me file a ticket to address this. 

https://github.com/NASA-AMMOS/aerie/issues/1350